frappe/email/doctype/email_account/email_account.py

```diff
@@ -475,9 +475,17 @@ def sendmail_config(self):
 		}
 
 	def get_smtp_server(self):
+		return self._smtp_server_instance
+
+	@functools.cached_property
+	def _smtp_server_instance(self):
```

Comment:
Funnily enough, this change alone is meat of whole PR. This remembers the SMTPServer and attaches it to email account doc. No need to cache anything separately for reusing. 

Email queue always uses cached documents and since cached documents are same exact object in case of single transaction, we can reuse connection without implementing any separate code for it at higher level.